src/api/api.ts

```diff
@@ -1 +1,53 @@
-export {};
+import axios, {AxiosRequestConfig} from 'axios';
+import {Alert} from 'react-native';
+import Config from 'react-native-config';
+
+const YelpApi = axios.create({
+  baseURL: Config.API_URL,
+});
+
+YelpApi.interceptors.request.use(
+  function (config) {
+    config.params = {
+      accept: 'application/json',
+      ...config.params,
+    };
+    config.headers.Authorization = `Bearer ${Config.TOKEN}`;
+    return config;
+  },
+  function (error) {
+    throw error;
+  },
+);
+
+export function Api<T>(
+  params: AxiosRequestConfig,
+  showError: boolean = true,
+): Promise<T> {
+  return YelpApi.request(params)
+    .then(res => {
+      res.data.stat = 'ok';
+      return res.data;
+    })
+    .catch(err => {
+      showError
+        ? Alert.alert(
+            'Error Occured',
+            err.response.data.error.description ?? 'Try Again After SomeTime',
+          )
+        : null;
+      err.showErrorScreen = !showError;
+      err.stat = 'fail';
+      if (err.response) {
+        err.description =
+          err.response.data.error.description ?? 'SomwThing Weird Happened';
```

Comment:
Spelling mistake. Also if we are setting description here. How come we are showing an alert before setting the message?